src/threading/cthread.cpp

```diff
@@ -173,7 +173,7 @@ static size_t global_get_num_threads() {
         terminate("The requested number of threads \""+std::string(str)
             +"\" is not a positive integer");
     }
-    auto num_threads = std::stoi(str);
+
     return std::stoi(str);
```

Comment:
By default `std::stoi` uses base10. You have to set explicitly set the third argument to 0 to get auto-detection of base. e.g.:
```
bcumming@arapiles:test > cat stoi.cpp 
#include <iostream>
#include <string>

int main(void) {
    std::cout << std::stoi("010") << "\n";
    std::cout << std::stoi("010", nullptr, 0) << "\n";
}
bcumming@arapiles:test > g++ stoi.cpp && ./a.out
10
8
```
the docs are here:
http://en.cppreference.com/w/cpp/string/basic_string/stol